site/documentation.markdown

```diff
@@ -6,27 +6,29 @@ isDocumentation: true
 
 # Documentation
 
+Resources marked with \[$$\] are paid.
+
 ## Introductory Books for Learning Haskell
 
-*   [Haskell from the Very Beginning](https://www.haskellfromtheverybeginning.com/)
+*   \[$$\] [Haskell from the Very Beginning](https://www.haskellfromtheverybeginning.com/)
 *   [Learn You a Haskell for Great Good!](http://learnyouahaskell.com/)
 *   [Real World Haskell](http://book.realworldhaskell.org/)
-*   [Haskell Programming from first principles](http://haskellbook.com)
-*   [Beginning Haskell](http://www.apress.com/9781430262503)
-*   [Thinking Functionally with Haskell](http://www.cambridge.org/us/academic/subjects/computer-science/programming-languages-and-applied-logic/thinking-functionally-haskell)
-*   [Programming in Haskell](http://www.cs.nott.ac.uk/~pszgmh/pih.html)
-*   [Haskell: The Craft of Functional Programming](http://www.haskellcraft.com/craft3e/Home.html)
-*   [The Haskell School of Music](http://euterpea.com/haskell-school-of-music/)
-*   [Get Programming with Haskell](https://www.manning.com/books/get-programming-with-haskell)
+*   \[$$\] [Haskell Programming from first principles](http://haskellbook.com)
+*   \[$$\] [Beginning Haskell](http://www.apress.com/9781430262503)
+*   \[$$\] [Thinking Functionally with Haskell](http://www.cambridge.org/us/academic/subjects/computer-science/programming-languages-and-applied-logic/thinking-functionally-haskell)
+*   \[$$\] [Programming in Haskell](http://www.cs.nott.ac.uk/~pszgmh/pih.html)
+*   \[$$\] [Haskell: The Craft of Functional Programming](http://www.haskellcraft.com/craft3e/Home.html)
+*   \[$$\] [The Haskell School of Music](http://euterpea.com/haskell-school-of-music/)
+*   \[$$\] [Get Programming with Haskell](https://www.manning.com/books/get-programming-with-haskell)
 
 ## Intermediate Haskell Books
 
 *   [Developing Web Applications with Haskell and Yesod](http://www.yesodweb.com/book)
-*   [Functional Design and Architecture](https://www.manning.com/books/functional-design-and-architecture)
-*   [Haskell in Depth](https://www.manning.com/books/haskell-in-depth)
+*   \[$$\] [Functional Design and Architecture](https://www.manning.com/books/functional-design-and-architecture)
+*   \[$$\] [Haskell in Depth](https://www.manning.com/books/haskell-in-depth)
 *   [Parallel and Concurrent Programming in Haskell](https://simonmar.github.io/pages/pcph.html)
```

Comment:
Is it really possible to read Parallel and Concurrent Programming in Haskell without payment? The link has a further "Read the book online" link to https://www.oreilly.com/library/view/parallel-and-concurrent/9781449335939/, but that requires an O'Reilly "free trial". I guess technically it doesn't require payment, but I don't know how much or for how long you can read without payment.

I *thought* the book was previously freely available online without paying anything, but if it was I can't find out how now ...

Reply:
Oh, it seems the contents of the book *is* available on that O'Reilly page, but nonetheless it keeps asking you to sign up for a free trial :)

Reply:
If you click on “Table of contents” and then on any section, you can read it seems you can read it at will, without logging in. Whether it is a bug or a feature I don’t know!